src/apps/monitoring/kadiska/mode/alerts.pm

```diff
@@ -0,0 +1,229 @@
+
+sub prefix_global_output {
+    my ($self, %options) = @_;
+    return 'Rules: ';
+}
+
+sub prefix_rules_output {
+    my ($self, %options) = @_;
+    return sprintf('Rule id: "%s", Rule name: "%s" ',
+        $options{instance_value}->{id},
+        $options{instance_value}->{name});
+}
+
+sub set_counters {
+    my ($self, %options) = @_;
+
+    $self->{maps_counters_type} = [
+        { name => 'global', type => 0, cb_prefix_output => 'prefix_global_output'},
+        { name => 'rules', type => 1, cb_prefix_output => 'prefix_rules_output' }
+    ];
+
+    $self->{maps_counters}->{global} = [
+        { label => 'rules-total-count', nlabel => 'rules.total.count', set => {
+                key_values => [ { name => 'total' }  ],
+                output_template => 'total rules: %s',
+                perfdatas => [ { template => '%d', min => 0 } ]
+            }
+        },
+        { label => 'rules-criticals-total-count', nlabel => 'criticals.total.count',
+            set
+                => {
+                key_values => [ { name => 'total_critical' }  ],
+                output_template => 'total criticals: %s',
+                perfdatas => [ { template => '%d', min => 0 } ]
+            }
+        },
+        { label => 'rules-warnings-total-count', nlabel => 'warnings.total.count',
+            set => {
+                key_values => [ { name => 'total_warning' }  ],
+                output_template => 'total warnings: %s',
+                perfdatas => [ { template => '%d', min => 0 } ]
+            }
+        },
+        { label => 'rules-nodata-total-count', nlabel => 'no.data.total.count',
+            set
+                => {
+                key_values => [ { name => 'total_nodata' }  ],
+                output_template => 'total no data: %s',
+                perfdatas => [ { template => '%d', min => 0 } ]
+            }
+        }
+    ];
+
+    $self->{maps_counters}->{rules} = [
+        { label => 'rule-ok-count',
+            type => 1,
+            set => {
+                key_values => [ { name => 'ok_count' } ],
+                output_template => 'ok count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        },
+        { label => 'rule-warning-count',
+            type => 1,
+            set => {
+                key_values => [ { name => 'warning_count' } ],
+                output_template => 'warning count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        },
+        { label => 'rule-critical-count',
+            type => 1,
+            set => {
+                key_values => [ { name => 'critical_count' } ],
+                output_template => 'critical count: %s',
+                display_ok => 0,
+                closure_custom_perfdata => sub { return 0; }
+            }
+        },
+        { label => 'rule-nodata-count',
+            type => 1,
```

Comment:
No need to set ```type => 1```. Why we don't have a perfdata for that counter ?

Reply:
This section was developped with this plugin as pattern :
https://github.com/centreon/centreon-plugins/blob/develop/src/hardware/devices/hms/netbiter/argos/restapi/mode/alarms.pm#L90

Does this use of perfdata is correct ?
```
perfdatas => [
        { template => '%s', min => 0, label_extra_instance => 1, instance_use => 'name' }
]
```